app/services/usps_exporter.rb

```diff
@@ -0,0 +1,46 @@
+class UspsExporter
+  OTP_MAX_VALID_DAYS = 30
+
+  def initialize(csv_file_path)
+    @csv_file_path = csv_file_path
+  end
+
+  def run
+    CSV.open(csv_file_path, 'wb', col_sep: '|') do |csv|
+      entries.each do |entry|
+        csv << make_entry_row(entry)
+      end
+    end
+    clear_entries
+  end
+
+  private
+
+  attr_reader :csv_file_path
+
+  def entries
+    UspsConfirmation.all.map(&:decrypted_entry)
+  end
+
+  def clear_entries
+    UspsConfirmation.destroy_all
```

Comment:
Also is the uploading of the CSV a separate step? If we really want to be careful, we shouldn't delete until we know that the CSV has been sent (maybe even acknowledged) on the other side?

I would include the CSV sending in this job if we get the chance